### PR TITLE
Add Trace V2 client and _gax

### PR DESCRIPTION
**Pull request by liyanhui1228**
For #4415

### CODE REVIEW

**Comment by liyanhui1228** [comment]
Friendly ping.

**Comment by dhermes** [comment]
@liyanhui1228 Who is the ping intended for?

**Comment by liyanhui1228** [comment]
Would like @lukesneeringer for initial review to make sure we agree on making Trace V2 API as the default.

File: trace/google/cloud/trace/__init__.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright 2017 Google LLC
+# Copyright 2017, Google LLC All rights reserved.
```
Comment (duggelz) [inline trace/google/cloud/trace/__init__.py]: From my reading of the OSPO guidelines, this was correct as it was, e.g.

`# Copyright 2017 Google LLC`

Unless you found another source that says something differently (very possible).

File: trace/google/cloud/trace/v1/_gax.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    :param client: The client that owns this API object.
+    """
+    def __init__(self, gax_api, client):
+        self._gax_api = gax_api
+        self.client = client
+
+    def patch_traces(self, project_id, traces, options=None):
+        """Sends new traces to Stackdriver Trace or updates existing traces.
+
+        :type project_id: str
```
Comment (duggelz) [inline trace/google/cloud/trace/v1/_gax.py]: Just to check, is there any reason we need to leave these with the old-style docstrings?  I.e. we have napoleon above like:

```
Args:
    start_time (:class:`~datetime.datetime`):
               The start time of the span. 
```

vs the `:type`, `:param` style here.

If there isn't a reason to keep them different, then we should file an issue against ourselves to go clean everything up to a single consistent style (some day).

Comment (dhermes) [reply trace/google/cloud/trace/v1/_gax.py]: @duggelz We've been "fine with both" for some time now. However we've privately discussed being OK with badgering Googlers to exclusively use Google-style docstrings.

Comment (liyanhui1228) [reply trace/google/cloud/trace/v1/_gax.py]: Changed to use napoleon style.

**Comment by liyanhui1228** [comment]
`from google.cloud import trace` will be using V2 as the default, and using V1 will be like `from google.cloud import trace_v1`. The V1 API includes both read and write methods, but V2 API only contains write methods.

**Pull request merged**